packages/app/src/vis-packs/core/compound/MappedCompoundMatrixVis.tsx

```diff
@@ -43,14 +44,20 @@ function MappedCompoundMatrixVis(props: Props) {
     getFormatter(field, notation)
   );
 
+  const { getExportURL } = useDataContext();
+  const selection = getSliceSelection(dimMapping);
+
   return (
     <>
       {toolbarContainer &&
         createPortal(
           <MatrixToolbar
-            dataset={dataset}
-            selection={getSliceSelection(dimMapping)}
             cellWidth={cellWidth}
+            isSlice={selection !== undefined}
+            getExportURL={
+              getExportURL &&
+              ((format) => getExportURL(dataset, selection, value, format))
```

Comment:
I'm now currying the `getExportURL` function from the mapped vis components to avoid passing `dataset` `selection` and `value` props to all the toolbars...